akka-stream-testkit/src/main/scala/akka/stream/testkit/StreamTestKit.scala

```diff
@@ -315,12 +474,49 @@ object TestSubscriber {
     def receiveWhile[T](max: Duration = Duration.Undefined, idle: Duration = Duration.Inf, messages: Int = Int.MaxValue)(f: PartialFunction[SubscriberEvent, T]): immutable.Seq[T] =
       probe.receiveWhile(max, idle, messages)(f.asInstanceOf[PartialFunction[AnyRef, T]])
 
+    /**
+     * Drains a given number of messages
+     */
     def receiveWithin(max: FiniteDuration, messages: Int = Int.MaxValue): immutable.Seq[I] =
       probe.receiveWhile(max, max, messages) {
         case OnNext(i) ⇒ Some(i.asInstanceOf[I])
         case _         ⇒ None
       }.flatten
 
+    /**
+     * Attempt to drain the stream into a strict collection (by requesting `Long.MaxValue` elements).
+     *
+     * '''Use with caution: Be warned that this may not be a good idea if the stream is infinite or its elements are very large!'''
+     */
+    def toStrict(atMost: FiniteDuration): immutable.Seq[I] = {
+      val deadline = Deadline.now + atMost
+      val b = immutable.Seq.newBuilder[I]
+
+      def checkDeadline(): Unit = {
+        if (deadline.isOverdue())
+          throw new TimeoutException(s"toStrict did not drain the stream within $atMost! Accumulated elements: ${b.result()}")
+      }
+
+      @tailrec def drain(): immutable.Seq[I] =
+        self.expectEvent() match {
+          case OnError(ex) ⇒
+            throw new AssertionError(s"toStrict received OnError($ex) while draining stream! Accumulated elements: ${b.result()}")
```

Comment:
perhaps not include the the full exception string in the message? Can it be a cause?